### PR TITLE
Tagging and sample data quide

### PR DESCRIPTION
**Pull request by shairyar**
Create a guide for tagging and sample data. Merge the language-specific docs into one guide.

- [x] Update Elixir code example to reflect the new 2.0 version.
- [x] Merge main and resolve the conflicts.
- [x] Update navigation on the start page and in the guide before/after
- [x] Link templates

### CODE REVIEW

File: source/application/tagging-sample-data.html.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
+---
+title: "Tagging and Sample Data"
+---
+
+You can supply extra context on errors and performance issues using tags and sample data.
```
Comment (tombruijn) [inline source/application/tagging-sample-data.html.md:5]: Should we expand the tagging topic to also mention tagging for metrics? A tagging system also exists for metrics.

Comment (shairyar) [reply source/application/tagging-sample-data.html.md:5]: I agree and we have tag for exceptions too

Comment (tombruijn) [reply source/application/tagging-sample-data.html.md:5]: Yes, tags for exceptions (e.g. set_error and send_error), are set on the sample data.

Comment (shairyar) [reply source/application/tagging-sample-data.html.md:5]: I am thinking if this guide should have a structure like filter data as well, where we can have separate sub guide for tag, sample data and then tag for metrics

Comment (tombruijn) [reply source/application/tagging-sample-data.html.md:5]: Yeah that could work. If it becomes too long, or you're having trouble guiding people through it, split it up.

File: source/guides/tagging-sample-data/sample-data.html.md [hunk 2]
```diff
@@ -0,0 +1,133 @@
+---
+title: "Adding sample data to a request"
+---
+
+Besides tags you can add more metadata to a transaction (or override default metadata from integrations such as Phoenix).
+
+## Ruby
+
+### `custom_data`
```
Comment (tombruijn) [inline source/guides/tagging-sample-data/sample-data.html.md:9]: Let's add a separate page for the custom data feature and explain it in more detail.

The rest of the page, should maybe be linked to from the Elixir and Node.js installation guides. As next steps for non-Phoenix and non-Plug apps.

File: source/application/integrations/webhooks.html.md [hunk 3]
```diff
@@ -264,14 +264,14 @@ There are multiple webhooks available. Which payloads the webhook receives can b
     <tr>
       <td><code>metadata</code></td>
       <td><code>Hash&lt;String, String&gt;</code></td>
-      <td>This includes user set metadata (see for <a href="/ruby/instrumentation/tagging.html">Ruby</a> / <a href="/elixir/instrumentation/tagging.html">Elixir</a>).</td>
+      <td>This includes user set metadata (see <a href="/guides/tagging-sample-data/">Tagging feature]</a>.</td>
```
Comment (tombruijn) [inline source/application/integrations/webhooks.html.md:267]:
```suggestion
      <td>This includes user set metadata (see <a href="/guides/tagging-sample-data/">Tagging feature</a>.</td>
```

File: data/navigation.yml [hunk 4]
```diff
@@ -276,9 +292,6 @@ sections:
               -
                 name: "Exception handling"
                 link: "/ruby/instrumentation/exception-handling.html"
-              -
```
Comment (tombruijn) [inline data/navigation.yml:279]: Let's set up redirects in the `netlify.toml` file for pages you move.

**Review by tombruijn — CHANGES_REQUESTED** [review]

**Comment by tombruijn** [comment]
The above was discussed in a call to give more context.

File: source/guides/tagging-sample-data/tagging-request.html.md [hunk 5]
```diff
@@ -0,0 +1,68 @@
+---
+title: "Adding tags to a request"
+---
+
+## Ruby
+
+### Limitations
```
[the remark this answers is not in the document]
Comment (shairyar) [reply source/guides/tagging-sample-data/tagging-request.html.md:7]: I have moved this at the bottom of the page.

File: source/guides/tagging-sample-data/tagging-request.html.md [hunk 6]
```diff
@@ -0,0 +1,68 @@
+---
+title: "Adding tags to a request"
+---
+
```
[the remark this answers is not in the document]
Comment (shairyar) [reply source/guides/tagging-sample-data/tagging-request.html.md:4]: I have added the introduction

File: source/guides/tagging-sample-data/index.html.md [hunk 7]
```diff
@@ -0,0 +1,29 @@
+---
```
[the remark this answers is not in the document]
Comment (shairyar) [reply source/guides/tagging-sample-data/index.html.md:1]: I have renamed` tagging-sample-data` directory to `custom-data` and updated its references throughout the docs.

File: source/application/tagging-sample-data.html.md [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+
+You can supply extra context on errors and performance issues using tags and sample data. 
+This can help to add information that is not already part of the request, session or environment parameters.
+
+
+!> **Warning**: Do not use tagging to send personal data such as names or email
+   addresses to AppSignal. If you want to identify a person, consider using a
+   user ID, hash or pseudonymized identifier instead. You can use
+   [link templates](/application/link-templates.html) to link them to your own
+   system.
```
[the remark this answers is not in the document]
Comment (shairyar) [reply source/application/tagging-sample-data.html.md:13]: I have removed this limit.

File: source/application/tagging-sample-data.html.md [hunk 9]
```diff
@@ -0,0 +1,30 @@
+---
```
[the remark this answers is not in the document]
Comment (shairyar) [reply source/application/tagging-sample-data.html.md]: File has been named.

File: source/application/tagging-sample-data.html.md [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+title: "Tagging and Sample Data"
+---
+
+You can supply extra context on errors and performance issues using tags and sample data. 
+This can help to add information that is not already part of the request, session or environment parameters.
```
[the remark this answers is not in the document]
Comment (shairyar) [reply source/application/tagging-sample-data.html.md]: I have added more description

**Review by tombruijn — APPROVED** [review]
I made some changes, I think it's good to merge for now.

File: source/guides/custom-data/sample-data.html.md [hunk 11]
```diff
@@ -1,32 +1,29 @@
...
 
-### Link templates
+```
+Appsignal.Span.set_sample_data(Appsignal.Tracer.root_span, "custom_data", %{foo: "bar"})
 
-Tags can also be used to create link templates. Read more about link templates in our [link templates guide](/application/link-templates.html).
+```
 
-## Sample Data
+## Node.js
```
[the remark this answers is not in the document]
Comment (shairyar) [reply source/guides/custom-data/sample-data.html.md:52]: Yes thats exactly what I had in my mind as this PR is becoming too long.

**Pull request merged**